app/javascript/controllers/currency_dropdown_controller.js

```diff
@@ -0,0 +1,58 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="dropdown"
+export default class extends Controller {
+  static targets = ["menu", "input", "label", "option"]
+
+  toggleMenu(event) {
+    event.stopPropagation(); // Prevent event from closing the menu immediately
+    this.repositionDropdown();
+    this.menuTarget.classList.toggle("hidden");
+  }
+
+  hideMenu = () => {
+    this.menuTarget.classList.add("hidden");
+  }
+
+  connect() {
+    document.addEventListener("click", this.hideMenu);
+  }
+
+  disconnect() {
+    document.removeEventListener("click", this.hideMenu);
+  }
```

Comment:
what do you think about binding the event to the element instead as follow:

```suggestion
  connect() {
    this.element.addEventListener("click", this.hideMenu.bind(this);
  }

  disconnect() {
    this.element.removeEventListener("click", this.hideMenu.bind(this);
  }
```

however i'm wondering if it would make more sense to add `this.hideMenu()` in the `selectOption` function instead.

Reply:
I believe the intention here is to capture clicks that occur outside of the dropdown so that the dropdown becomes closed.

Binding the event to the element wouldn't capture those clicks.

Reply:
Makes sense!

I saw that this is done in the js `modal_controller` like this:
```js
click_outside(e) {
    if (e.target === this.element) {
      this.element.close();
    }
  }
 ``` 
 
 stimulus build in events could also be used like:
 ```js
 data-action="click@window->currency-dropdown#hideMenu"
 ```
 
 but already looks good to go as it is!